apps/solution/src/components/people/people.tsx

```diff
@@ -1,18 +1,47 @@
 import { Grid } from '@libs/design';
+import { Dispatch, SetStateAction, useState } from 'react';
 import Person from '../person/person';
 import usePeople from '../person/usePeople';
 
 import style from './people.module.scss';
 
+function normalize(text: string) {
+  return text.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "")
+}
+
 export function People() {
   const people = usePeople();
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const filteredPeople = people?.filter(person => {
+    const searchTermNormalized: string = normalize(searchTerm)
+    return normalize(person.lastname).includes(searchTermNormalized) || normalize(person.firstname).includes(searchTermNormalized)
+  })
 
   return (
-    <Grid className={style.container}>
-      {people?.map((person) => <Person key={person.id} person={person} />) ||
-        'Loading ...'}
-    </Grid>
+    <>
+      <SearchBar searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
+      <Grid className={style.container}>
+        {filteredPeople?.map((person) => <Person key={person.id} person={person} />) ||
+          'Loading ...'}
+      </Grid>
+    </>
   );
 }
 
+
+export interface SearchBarProps {
+  searchTerm: string;
+  setSearchTerm: Dispatch<SetStateAction<string>>;
+}
+function SearchBar({ searchTerm, setSearchTerm }: SearchBarProps) {
```

Comment:
Probably just for educational purpose only, but I would rename `setSearchTerm` into `onSearchTermChange` or anything else explaining it can be any callback function, not always the dispatcher.

Reply:
I did change it, it will indeed allow the student to understand that this can be anything really.